### DISCUSSION TITLE
Are URL-pattern settings supposed to affect *all* requests made from this tab?

### DISCUSSION THREAD

**Discussion by max-baz** [qutebrowser/qutebrowser#6344] [Q&A] [answered]
Here's what I'm seeing: I set by default `content.cookies.accept` to `no-3rdparty`, and it works well for me on all websites except one - Microsoft Teams (of course 😐), which refuses to load (it shows a prompt asking to refresh the page, again and again - and console is full of 401 errors).

So I thought I would trick it by additionally setting `content.cookies.accept = all` only for `*://teams.microsoft.com` domain.

However it doesn't solve the issue, I'm still getting this prompt.

If I simply do `:set content.cookies.accept all` (without any URL pattern), then Teams will load and work fine.

I expected that the intended behavior of `:set --pattern *://teams.microsoft.com content.cookies.accept all` would be such that if I open `https://teams.microsoft.com`, _on this tab_ all the cookies are accepted. But it doesn't seem to be the case.

Is my expectation incorrect? Does that mean that a "proper" way to solve the issue is to hunt down all domains, from which Teams tab is making a request?

**Comment by The-Compiler** [comment, marked as answer]
From the [settings documentation](https://github.com/qutebrowser/qutebrowser/blob/master/doc/help/settings.asciidoc#content.cookies.accept) for `content.cookies.accept`:

> If this setting is used with URL patterns, the pattern gets applied to the origin/first party URL of the page making the request, not the request URL.

URL patterns in settings are entirely unrelated to tabs, though. If you set `content.cookies.accept = all` for `*://teams.microsoft.com`, then any third-party cookie will be accepted if it's being set on `teams.microsoft.com`, no matter what tab that is.

In this case, I suspect what you're seeing is actually [[QTBUG-90231] [REG 5.15.1 -> 5.15.2] firstPartyUrl for cookie filter has subdomains stripped](https://bugreports.qt.io/browse/QTBUG-90231) - so you might need to use just `microsoft.com` (or `*.microsoft.com`) in your pattern.

If you run with `--debug --logfilter network --debug-flag log-cookies` in a terminal, you can find out more about what's going on.

**Reply by max-baz** [reply]
That was exactly right, thank you very much 🙂
